### ISSUE TITLE
All available Cronet providers are disabled. A provider should be enabled before it can be used.

### ISSUE THREAD

**Issue by abdullahalshafi** [yoobi/exoplayer-kotlin#10]
Hi, Thank you very much for your repo. 

Getting this crash in firebase console. Can you please help?

Happens here.

httpDataSourceFactory = CronetDataSource.Factory(
                CronetEngine.Builder(context).build(),
                Executors.newSingleThreadExecutor()
            )

I am using:  implementation 'com.google.android.exoplayer:extension-cronet:2.18.1'

**Comment by yoobi** [comment]
Hello,

Could you share the code of your implementation and the exact error ?

**Comment by abdullahalshafi** [comment]
Thank you for your prompt response. 

I am using your downloadvideo.

Here is the crash log:

Caused by java.lang.RuntimeException: All available Cronet providers are disabled. A provider should be enabled before it can be used.
       at org.chromium.net.CronetEngine$Builder.getEnabledCronetProviders(CronetEngine.java:365)
       at org.chromium.net.CronetEngine$Builder.createBuilderDelegate(CronetEngine.java:327)
       at org.chromium.net.CronetEngine$Builder.<init>(CronetEngine.java:75)
       at com.activity.videoplayer.common.DownloadUtil.getHttpDataSourceFactory(DownloadUtil.kt:42)
       at com.activity.videoplayer.common.DownloadUtil.ensureDownloadManagerInitialized(DownloadUtil.kt:121)
       at com.activity.videoplayer.common.DownloadUtil.getDownloadTracker(DownloadUtil.kt:83)
       at com.activity.videoplayer.VideoPlayerActivity.init(VideoPlayerActivity.kt:202)
       at com.activity.videoplayer.VideoPlayerActivity.onCreate(VideoPlayerActivity.kt:196)
       at android.app.Activity.performCreate(Activity.java:8403)
       at android.app.Activity.performCreate(Activity.java:8380)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1323)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4264)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4495)
       at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:93)
       at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:149)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:103)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2706)
       at android.os.Handler.dispatchMessage(Handler.java:109)
       at android.os.Looper.loop(Looper.java:228)
       at android.app.ActivityThread.main(ActivityThread.java:9100)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:614)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1129)
       
Basically I call DownloadUtil.getDownloadTracker(this).addListener(this) from my player activity and this crash happens in DownloadUtil class.

**Comment by yoobi** [comment]
Hello,

My guess is that the phones who's having crash don't have Google Play Services and so you get this crash, it's out of the scope of this repo.
I've started the app on emulator which doesn't have Google Play Services and it crash with this log, while I've run the app correctly on an emulator which have Google Play Services .

You might wanna check into https://developer.android.com/develop/connectivity/cronet/start#set-up

**Comment by abdullahalshafi** [comment]
Thank you very much for your help. One more thing if you could help me out with, is there any way to know if Cronet is available or not?

**Comment by yoobi** [comment]
Yep this is confirmed in the doc https://exoplayer.dev/network-stacks.html#choosing-a-network-stack

Cronet requires Google Play Services. To know if Cronet is available you can check if phone has google play services I guess
